test/e2e/config/azure-dev.yaml

```diff
@@ -92,7 +92,7 @@ providers:
       replacements:
       - old: "imagePullPolicy: Always"
         new: "imagePullPolicy: IfNotPresent"
-    - name: v1.9.99 # "vNext"; use manifests from local source files
+    - name: v1.10.99 # "vNext"; use manifests from local source files
```

Comment:
we also need to update the test metadata as described in https://capz.sigs.k8s.io/developers/releasing.html#update-test-capz-provider-metadatayaml-skip-for-patch-releases